.travis/release.sh

```diff
@@ -9,10 +11,22 @@ if [ "${BUILD}" = "deploy" ]; then
 
 # Deploy to ossrh has already been done with the usual build. See build-deploy.sh
 
-# The site has been built before, the files have already been uploaded to sourceforge.
-# Since this is a release, making the binary the new default file...
-curl -H "Accept: application/json" -X PUT -d "default=windows&default=mac&default=linux&default=bsd&default=solaris&default=others" \
-     -d "api_key=${PMD_SF_APIKEY}" https://sourceforge.net/projects/pmd/files/pmd/${RELEASE_VERSION}/pmd-bin-${RELEASE_VERSION}.zip
+(
+    # disable fast fail, exit immediately, in this subshell
+    set +e
```

Comment:
If I understand right, this means that if curl returns a non-zero code, then the subshell doesn't immediately end and you can output your debug message. But when does the subshell terminate then?

Reply:
The subshell ends with the closing parentheses  `)` . If one command fails within the subshell, then simply the next command is executed. Everytime you use parentheses, a new subshell is spawned.